validator/README.md

```diff
@@ -50,11 +50,12 @@ the edges. Below are instructions for Linux Ubuntu 14.
 
 Install these packages using apt-get:
 
-* npm
-* openjdk-7-jre
-* protobuf-compiler
-* python-protobuf
-* python2.7
+* `npm`
+* `openjdk-7-jre`
+* `protobuf-compiler`
+* `python2.7`
+
+Then use pip to `pip install protobuf`.
```

Comment:
I just remembered why we switched away from `pip`: caching on Travis doesn't work for non python projects (ours is a nodejs project)

A potential alternative might be to use an updated apt source. https://docs.travis-ci.com/user/installing-dependencies/#adding-apt-sources

Any chance you can explore that path?

Reply:
I did briefly explore it before I went with `pip` - the issue is not (just) that Xenial's apt sources don't have protobuf 3, it seems it hasn't been backported to Xenial at all, per protocolbuffers/protobuf#1799.

Are you sure `pip` is not being cached? I thought this line would do that:

https://github.com/ampproject/amphtml/blob/2a5205e2de99a99c2ccca0b50b4355fbb20f277a/.travis.yml#L94

[This repo](https://github.com/nickstenning/travis-pip-cache) also makes me think that's the correct approach to cache `pip`.

Reply:
Fair enough!